packages/flame/lib/src/components/core/component_tree_root.dart

```diff
@@ -127,6 +127,23 @@ class ComponentTreeRoot extends Component {
       }
     });
   }
+
+  @mustCallSuper
+  @internal
+  void registerKey(ComponentKey key, Component component) {
+    _index[key] = component;
+  }
+
+  @mustCallSuper
+  @internal
+  void unregisterKey(ComponentKey key) {
+    _index.remove(key);
+  }
+
+  T? findByKey<T extends Component>(ComponentKey key) {
```

Comment:
Maybe there could be a method that finds it by name too?

Reply:
Done!